### PR TITLE
test: add test case based on celestia-node usage

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/rsmt2d/issues/238

cc: @walldiss @Wondertan

### CODE REVIEW

File: rsmt2d_test.go [hunk 1]
```diff
@@ -157,3 +158,57 @@ func TestEdsRepairTwice(t *testing.T) {
...
+
+	// Populate EDS with 1/4 of chunks using SetCell
+	err = eds.SetCell(0, 0, exampleEds.GetCell(0, 0))
+	require.NoError(t, err)
+	err = eds.SetCell(0, 1, exampleEds.GetCell(0, 1))
+	require.NoError(t, err)
+	err = eds.SetCell(1, 0, exampleEds.GetCell(1, 0))
+	require.NoError(t, err)
+	err = eds.SetCell(1, 1, exampleEds.GetCell(1, 1))
+	require.NoError(t, err)
```
Comment (rootulp) [inline rsmt2d_test.go:184]: Considered refactoring to

```go
	for row := 0; row < edsWidth/2; row++ {
		for col := 0; col < edsWidth/2; col++ {
			err = eds.SetCell(uint(row), uint(col), exampleEds.GetCell(uint(row), uint(col)))
			require.NoError(t, err)
		}
	}
```

but I think the existing implementation is easier to read.

File: rsmt2d_test.go [hunk 2]
```diff
@@ -157,3 +158,57 @@ func TestEdsRepairTwice(t *testing.T) {
...
+	rowRoots, err := exampleEds.RowRoots()
+	require.NoError(t, err)
+	colRoots, err := exampleEds.ColRoots()
+	require.NoError(t, err)
+
+	// Repair the EDS
+	err = eds.Repair(rowRoots, colRoots)
+	assert.NoError(t, err)
+
+	assert.Equal(t, exampleEds.GetCell(3, 3), eds.GetCell(3, 3))
```
Comment (staheri14) [inline rsmt2d_test.go:198]: [Question] I'd suggest checking all the cells for equality (like through a loop).

**Review by staheri14 — APPROVED** [review]
LGTM!

File: rsmt2d_test.go (hunk 2 above)
Comment (rootulp) [reply rsmt2d_test.go]: Resolved via 11bb1c0a7c3531e6813fca015496234ba1f47988

**Review by staheri14 — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**